package/base-files/files/etc/init.d/led

```diff
@@ -49,6 +51,11 @@ load_led() {
 	[ -e /sys/class/leds/${sysfs}/brightness ] && {
 		echo "setting up led ${name}"
 
+
+		max_brightness="$(cat /sys/class/leds/${sysfs}/max_brightness)"
+		[ "$brightness" -lt "0" ] || [ "$brightness" -gt "$max_brightness" ] &&
+			brightness="${max_brightness}"
+
```

Comment:
why not make it a percentage? 0 - 100. so that it behaves the same on every platform. e.g. `$((brightness * max_brightness / 100))` or something.

Reply:
That's a good idea as it would simplify the validation in luci and make it straightforward

Reply:
Actually, I think it is better to leave it as-is. If we want to be able to correctly configure RGB leds/indicator, it is way easier to use the 0 - 255 value than a %,.

Reply:
Well, then why care about max brightness? also, realize that not every led has the max of 255. could be 16, could be 100, could be anything really. So by making the user set the 'exact value' you make it much harder to be generic.

Instead of 0 - 100% you can also do 0 - 1000 or 0 - 100000 so you can do float to int more easily.

A nother option of course is to let luci handle this completely, but having a generic interface is usually nicer.

Reply:
I think that's a bit overkill don't you agree?

Also, if your 100 isn't max, you can always do `if [ "${brightness}" -eq 100 ]; cat max_brightness > /sys` (and the same for -eq 0) (or solve it in math :p)

So then the only issue remains, 'what if I want to do brightness `123` and then, yeah sure this is the wrong interface :) but then you'd have to come up with a very complex interface :p 

Though if your percentage is fine grained enough; it shouldn't ever be an issue ...

Reply:
Yes, that is probably overkill, I just wanted to see all the solutions before doing the change.
I'll do the proper modifications later today or tomorrow. Thanks a lot for your input, I really appreciate the feedback.

Reply:
So, I did a couple of tests and to get enough control to use RGB leds properly (which is the main reason why I'm doing this) , it needs to be 0 - 1000. It feels a bit wrong to me. 
The more I think about it, the more I would leave as-is, but I might be totally wrong. Is there a way to get more opinions on that?

Reply:
While more opinions are certainly good :) I don't however see how this feels so wrong.

If we where talking about  0 - 1, then `0,1` or `0,01` or `0,001` would just mean 'more resolution' right. So now you are doing it on the integer spectrum instead :) doing exactly the same. I think a resolution of 1000 should cover current 8 bit leds, and potential 10 and 12 bit future leds as well? 16 bit brightness steps would be curious though :) (just from a point of 'lets settle on an interface pov).

Reply:
> If we where talking about 0 - 1, then 0,1 or 0,01 or 0,001 would just mean 'more resolution' right. So now you are doing it on the integer spectrum instead

From that point of view, it does indeed make a lot of sense. What was feeling wrong to me is that it is less intuitive then 0 - 100 or 0 - max. You cannot know without a doc or example what the scale is and it can be hard to guess it.

I'll do the changes (for real this time) and let's see what comes out of it.

Reply:
done